Chat_App/Chat_App/UI/ViewController/ViewController.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  ViewController.swift
+//  Chat_App
+//
+//  Created by Gi Oo on 24.04.23.
+//
+
+import UIKit
+
+class ViewController: UIViewController {
+
+    // MARK: Properties
+    private lazy var topMessageHistoryView = MessageHistoryView()
+    private lazy var dividerView = UIView()
+    private lazy var bottomChatHistoryView = MessageHistoryView()
+    private lazy var switchButtonView = SwitchButtonView()
+    private var statusBarStyle: UIStatusBarStyle = .darkContent
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        addSubviews()
+        setUpLayoutConstraints()
+        setUpDividerView()
+        setUpSwitchButtonViewToggle()
+    }
+
+    override var preferredStatusBarStyle: UIStatusBarStyle {
+        return statusBarStyle
+    }
+
+    private func setUpMessageViewTextColor(with color: UIColor) {
+        topMessageHistoryView.setUpTypingComponentView(with: color)
+        bottomChatHistoryView.setUpTypingComponentView(with: color)
+    }
+
+    private func setUpDividerView() {
+        dividerView.backgroundColor = AppColors.dividerViewColor
+    }
+
+    private func setUpSwitchButtonViewToggle() {
+        switchButtonView.onToggle = { [weak self] isOn in
+            if isOn {
+                self?.view.backgroundColor = AppColors.darkModeColor
+                self?.statusBarStyle = .lightContent
+                self?.setUpMessageViewTextColor(with: .white)
+            } else {
+                self?.view.backgroundColor = .white
+                self?.statusBarStyle = .darkContent
+                self?.setUpMessageViewTextColor(with: .black)
+            }
```

Comment:
if ის და else ის ბლოკებში რაც გიწერია არის იდენტური. აკეთებ კოდის დუბლირებას ამიტომ ცალკე ფუნქციაში გაიტანე ეგ ლოგიკები. ფუნქციას ექნება პარამეტრი მაგალითად isOn და მაგის მიხედვით გადაწყვეთ დანარჩენ ლოგიკებს

Reply:
// MARK: - SetUpMessageViewTextColor
    private func setUpMessageViewTextColor(with color: UIColor) {
        topMessageHistoryView.setUpTypingComponentView(with: color)
        bottomChatHistoryView.setUpTypingComponentView(with: color)
    }

    // MARK: - SetUpSwitchButtonViewToggle
    private func setUpSwitchButtonViewToggle() {
        let isDarkMode = UserDefaults.standard.bool(forKey: "AppTheme")
        updateAppearance(isDarkMode: isDarkMode)

        switchButtonView.onToggle = { [weak self] isOn in
            UserDefaults.standard.set(isOn, forKey: "AppTheme")
            self?.updateAppearance(isDarkMode: isOn)
        }
    }

    private func updateAppearance(isDarkMode: Bool) {
        if isDarkMode {
            view.backgroundColor = AppColors.darkModeColor
            statusBarStyle = .lightContent
            setUpMessageViewTextColor(with: .white)
        } else {
            view.backgroundColor = .white
            statusBarStyle = .darkContent
            setUpMessageViewTextColor(with: .black)
        }
        setNeedsStatusBarAppearanceUpdate()
    }
    
    SetUpSwitchButtonViewToggle ფუნქციას ვიძახებ viewDidLoad ში  და რამდენად სწორია?

Reply:
SetUpSwitchButtonViewToggle ფუნქციას რომ იძახებ viewDidLoad ეგ სწორია, მაგრამ SetUpSwitchButtonViewToggle ფუნქციაში უნდა ეწეროს მხოლოდ onToggle ქლოჟერი იმიტომ, რომ დარქმოუდია თუ არა საწყის ეტაპზე არ ეკუთვის სვიჩერის სეთაფს. ეგ ორი ხაზი გაიტანე ცალკე ფუნქციაში და ისე გამოიძახე viewDidLoad ში.

რაც შეეხება updateAppearance ფუნქციას მადნ if else ის ნაცვლად ტერნარული ოპერატორი გამოიყენე